src/io/flutter/run/LaunchState.java

```diff
@@ -119,25 +119,31 @@ protected RunContentDescriptor launch(@NotNull ExecutionEnvironment env) throws
     }
 
     final Project project = getEnvironment().getProject();
-    final FlutterDevice device = DeviceService.getInstance(project).getSelectedDevice();
-
-    // Flutter web does not yet support device objects yet, hence the null as the indicator that this might be a flutter web project.
-    if (device == null) {
-      boolean isFlutterWeb = false;
-      final String filePath = ((SdkRunConfig)runConfig).getFields().getFilePath();
-      if (filePath != null) {
-        final MainFile main = MainFile.verify(filePath, project).get();
-        final PubRoot root = PubRoot.forDirectory(main.getAppDir());
-        if (root != null) {
-          isFlutterWeb = FlutterUtils.declaresFlutterWeb(root.getPubspec());
-        }
-      }
-      if (!isFlutterWeb) {
-        showNoDeviceConnectedMessage(project);
-        return null;
+    @Nullable FlutterDevice device = DeviceService.getInstance(project).getSelectedDevice();
+
+    boolean isFlutterWeb = false;
+    final String filePath = ((SdkRunConfig)runConfig).getFields().getFilePath();
```

Comment:
Do you need to normalize here to ensure windows is supported, or has that already happened in getFilePath?

Reply:
This just restructures the code on the left to remove a level of nesting - it's fundamentally the same as the previous code.